Data/Migrations/20190219002519_AddAnnotationTable.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Data.Migrations
+{
+    public partial class AddAnnotationTable : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Annotations",
+                columns: table => new
+                {
+                    PuzzleID = table.Column<int>(nullable: false),
+                    TeamID = table.Column<int>(nullable: false),
+                    Key = table.Column<int>(nullable: false),
+                    Version = table.Column<int>(nullable: false),
+                    Contents = table.Column<string>(maxLength: 255, nullable: false),
+                    Timestamp = table.Column<DateTime>(nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Annotations", x => new { x.PuzzleID, x.TeamID, x.Key });
+                    table.ForeignKey(
+                        name: "FK_Annotations_Puzzles_PuzzleID",
+                        column: x => x.PuzzleID,
+                        principalTable: "Puzzles",
+                        principalColumn: "ID",
+                        onDelete: ReferentialAction.Cascade);
```

Comment:
Having cascading deletes for both Puzzle and Team is going to break when it meets up with #243 (which I'll be merging tonight) . You'll need to disable cascading deletes for one of them and manually delete annotations for it. See #243 for examples with PuzzleStatePerTeam.

Reply:
I don't understand.